uiowa/uiowa_profile
# Initial media/WYSIWYG implementation.

Starting a PR to track discussion, issues, etc. Also makes integration easier. This address https://github.com/uiowa/uiowa_profile/projects/1#card-7754465 for D8 MVP.

## Feature sets in this PR
- Upload images and files as media using media_entity_browser.
- Adds the ability to embed image and file media assets via entity_embed into the wysiwyg.
- Has ability to caption images through WYSIWYG.
- Adds logic to restrict display plugins based upon a media bundle as well as cleans up form items for file media types.
- Adds logical view modes for displaying image media entities (media:image_small, media:image_medium, media:image_large).
- Requires adding alt text to image media entities when upload.
- Creates a media library to reuse existing media.
- Creates a filtered html text format.
- Adds ability to scale and crop images based on focal point.
- Implements better internal linking via wysiwyg link button, including autocomplete and additional attributes.
- Creates new form displays to clean up core's media entity form when uploading via media_entity_browser.

## How to test
1. Edit `composer.json` file located at the application root.
2. Update uiowa/uiowa_profile to `dev-uiowa_media_wysiwyg`.
3. Save and close the file.
4. From the command line run: `composer update uiowa/uiowa_profile`.
5. Run a site-install: `drush si`.
6. Check features sets (see above) and other general media and wysiwyg functionality.
